internal/volumes/volumes.go

```diff
@@ -374,7 +375,11 @@ func GetCacheMount(args []string, _ storage.Store, _ string, additionalMountPoin
 			return newMount, nil, fmt.Errorf("no stage found with name %s", fromStage)
 		}
 		// path should be /contextDir/specified path
-		newMount.Source = filepath.Join(mountPoint, filepath.Clean(string(filepath.Separator)+newMount.Source))
+		newSrc, err := securejoin.SecureJoin(mountPoint, filepath.Clean(string(filepath.Separator)+newMount.Source))
```

Comment:
Try making this align more with what GetBindMount() is doing around line 198, so that they work the same.